apps/condo/bin/prepare-review-deploy.js

```diff
@@ -0,0 +1,112 @@
+    const { distDir, keystone, apps } = require(resolved)
+    const graphqlIndex = apps.findIndex(app => app instanceof GraphQLApp)
+    await keystone.prepare({ apps: [apps[graphqlIndex]], distDir, dev: true })
+    await keystone.connect()
+    context = await keystone.createContext({ skipAccessControl: true })
+
+    return context
+}
+
+// https://review-custom-name-***.*** -> "review-custom-name-"
+function getReviewPrefix ({ reviewDomain }) {
+    const condoDomain = conf['CONDO_DOMAIN']
+    // controlled reviewDomain
+    // nosemgrep: javascript.lang.security.audit.detect-non-literal-regexp.detect-non-literal-regexp
+    const findPrefixRegex = new RegExp(`https://(review-[a-zA-Z0-9-]+?-)\\w+\\.${reviewDomain.replaceAll('.', '\\.')}`)
+    const match = condoDomain.match(findPrefixRegex)
+    return match?.[1] || ''
+}
+
+let REVIEW_PREFIX = ''
+function updateUrls (obj, { devDomain, reviewDomain }) {
+    // controlled devDomain
+    // nosemgrep: javascript.lang.security.audit.detect-non-literal-regexp.detect-non-literal-regexp
+    const urlRegex = new RegExp(`https://([a-zA-Z0-9-]+)\\.${devDomain.replaceAll('.', '\\.')}(\\/[^\\s]*)?`, 'g')
+
+    function traverse (current) {
+        if (typeof current === 'string') {
+            return current.replace(urlRegex, (match, subdomain, path = '') => {
+                return `https://${REVIEW_PREFIX}${subdomain}.${reviewDomain}${path}`
+            })
+        } else if (Array.isArray(current)) {
+            return current.map(item => traverse(item))
+        } else if (typeof current === 'object' && current !== null) {
+            const result = {}
+            for (const key in current) {
+                result[key] = traverse(current[key])
+            }
+            return result
+        } else {
+            return current
+        }
+    }
+
+    return traverse(obj)
+}
+
+/**
+ * Changes all urls in oidcClient.payload from devDomain to reviewDomain with prefix
+ * @return {Promise<void>}
+ */
+async function migrateOidcClientUrlsToReview ({ devDomain, reviewDomain }) {
+    if (!REVIEW_PREFIX) throw new Error('Don\'t have review prefix!')
+
+    const context = await getContext()
+
+    const oidcClients = await OidcClient.getAll(context, {
+        deletedAt: null,
+    }, 'id clientId payload dv sender { dv fingerprint }')
+
+    const updatedOidcClients = oidcClients.map((client) => {
+        return {
+            ...client,
+            payload: updateUrls(client.payload, { devDomain, reviewDomain }),
+        }
+    })
+
+    for (const oidcClient of updatedOidcClients) {
+        try {
+            await OidcClient.update(context, oidcClient.id, {
+                payload: oidcClient.payload,
+                dv: oidcClient.dv,
+                sender: oidcClient.sender,
+            })
+        } catch (error) {
+            console.error(error)
+        }
+    }
+}
+
+
+async function main (args) {
+    const [devDomain, reviewDomain] = args
```

Comment:
Not obvious, that this one is roots, not domains itself...Got it only after visiting to helms

Reply:
renamed